routes/todos.js

```diff
@@ -0,0 +1,62 @@
+// import { v4 as uuidv4 } from 'uuid';
+const { v4: uuidv4 } = require('uuid');
+var express = require('express');
+var router = express.Router();
+var moment = require('moment');
+
+
+
+var todos = [{"id": "1", "name": "test", "date": "03/04/2020"}, {"id": "2", "name": "test2", "date": "03/04/2020"}]
+
+/* GET home page. */
+router.get('/', function(req, res, next) {
+  res.send(todos);
+});
+
+router.get('/:id', function(req, res, next) {
+  var todo = todos.find(element => element.id === req.params.id);
+  if (todo) {
+    res.status(200).send(todo);
+  } else {
+    res.status(404).send();
+  }
+});
+
+router.post('/', function(req, res, next) {
+  var name = req.body.name;
+  var id = uuidv4();
+  var date = moment().format();
+
+  var todo = {id, name, date}
+
+  todos.push(todo);
+
+  res.send(todo);
+});
+
+router.put('/:id', function(req, res, next) {
+  var name = req.body.name;
+  var index = todos.findIndex(element => element.id === req.params.id);
+  if (index !== -1) {
+    todos[index].name = name;
+    res.send(todos[index]);
+  } else {
+    res.status(404).send();
+  }
+})
+
+router.delete('/:id', function(req, res, next) {
+  var id = req.params.id;
+
+  var arrayLength = todos.length;
+  todos = todos.filter(element => element.id !== id)
```

Comment:
We can use : `var index = todos.findIndex(element => element.id === req.params.id);` then slice the array.
And it will allow us to use a const in place of var for `todos`.